src/main/java/org/apache/sysml/api/DMLScript.java

```diff
@@ -216,7 +234,200 @@ public static void main(String[] args)
 			// In case of DMLScriptException, simply print the error message.
 			System.err.println(e.getMessage());
 		}
-	} 
+	}
+
+	/**
+	 * Parses command line arguments to create a {@link DMLOptions} instance with the correct options
+	 * @param args	arguments from the command line
+	 * @param options	an {@link Options} instance containing the options that need to be parsed
+	 * @return an instance of {@link Options} that contain the correct {@link Option}s.
+	 * @throws org.apache.commons.cli.ParseException if there is an incorrect option specified in the CLI
+	 */
+	public static DMLOptions parseCLArguments(String[] args, Options options) throws org.apache.commons.cli.ParseException {
+
+		CommandLineParser clParser = new PosixParser();
+		CommandLine line = clParser.parse(options, args);
+
+		DMLOptions dmlOptions = new DMLOptions();
+		dmlOptions.help = line.hasOption("help");
+		dmlOptions.scriptType = line.hasOption("python") ? ScriptType.PYDML : ScriptType.DML;
+		dmlOptions.debug = line.hasOption("debug");
+		dmlOptions.gpu = line.hasOption("gpu");
+		if (dmlOptions.gpu) {
+			String force = line.getOptionValue("gpu");
+			if (force != null) {
+				if (force.equalsIgnoreCase("force")) {
+					dmlOptions.forceGPU = true;
+				} else {
+					throw new org.apache.commons.cli.ParseException("Invalid argument specified for -gpu option");
+				}
+			}
+		}
+		if (line.hasOption("exec")){
+			String execMode = line.getOptionValue("exec");
+			if (execMode != null){
+				if (execMode.equalsIgnoreCase("hadoop")) dmlOptions.execMode = RUNTIME_PLATFORM.HADOOP;
+				else if (execMode.equalsIgnoreCase("singlenode")) dmlOptions.execMode = RUNTIME_PLATFORM.SINGLE_NODE;
+				else if (execMode.equalsIgnoreCase("hybrid")) dmlOptions.execMode = RUNTIME_PLATFORM.HYBRID;
+				else if (execMode.equalsIgnoreCase("hybrid_spark")) dmlOptions.execMode = RUNTIME_PLATFORM.HYBRID_SPARK;
+				else if (execMode.equalsIgnoreCase("spark")) dmlOptions.execMode = RUNTIME_PLATFORM.SPARK;
+				else throw new org.apache.commons.cli.ParseException("Invalid argument specified for -exec option, must be one of [hadoop, singlenode, hybrid, hybrid_spark, spark]");
+			}
```

Comment:
I've removed Explain.parseExplainType. The reason for this is that parsing should not be used anywhere other than through the CLI. For the tests, for example, I will have it so that it sends in an instance of `DMLOptions` instead of a string array of `args` with all the systemml options.

In any case, my preference is towards this.